src/startup/ModeCLIParser.cpp

```diff
@@ -0,0 +1,158 @@
+            string("The FASTQ file which shall be indexed or - for stdin.") +
+            "The FASTQ file can also reside in an S3 bucket. Enter the filename here like s3:<filename> and set the"
+            " bucket with --bucket.",
+            true,
+            "-",
+            cmdLineParser);
+}
+
+_StringValueArg ModeCLIParser::createS3CredentialsFileArg(CmdLine *cmdLineParser) const {
+    return _makeStringValueArg(
+            "", "s3CredentialsFile",
+            string("In alternative to setting the bucket and the endpoint via command line, you can also specify ") +
+            "a configuration file. In this, you can also set proxy settings etc. Create a file with the s3init mode " +
+            "if you don't have one. The file format is described on the project website.",
+            false,
+            "", cmdLineParser);
+}
+
+_StringValueArg ModeCLIParser::createS3ConfigFileArg(CmdLine *cmdLineParser) const {
+    return _makeStringValueArg(
+            "", "s3ConfigFile",
+            string("In alternative to setting the bucket and the endpoint via command line, you can also specify ") +
+            "a configuration file. In this, you can also set proxy settings etc. Create a file with the s3init mode " +
+            "if you don't have one. The file format is described on the project website.",
+            false,
+            "", cmdLineParser);
+}
+
+_StringValueArg ModeCLIParser::createS3ConfigFileSectionArg(CmdLine *cmdLineParser) const {
+    return _makeStringValueArg(
+            "", "s3ConfigSection",
+            string("In alternative to setting the bucket and the endpoint via command line, you can also specify ") +
+            "a configuration file. In this, you can also set proxy settings etc. Create a file with the s3init mode " +
+            "if you don't have one. The file format is described on the project website.",
+            false,
+            "default",
+            cmdLineParser);
+}
+
+_SwitchArg ModeCLIParser::createForceOverwriteSwitchArg(CmdLine *cmdLineParser) const {
+    return _makeSwitchArg(
+            "w",
+            "forceOverwrite",
+            "Allow the indexer to overwrite an existing index file.",
+            cmdLineParser);
+}
+
+tuple<shared_ptr<UnlabeledValueArg<string>>, shared_ptr<ValuesConstraint<string>>>
+ModeCLIParser::createAllowedModeArg(const string &mode, CmdLine *cmdLineParser) const {
+    vector<string> allowedMode{mode};
+    auto allowedModesConstraint = make_shared<ValuesConstraint<string>>(allowedMode);
+    auto arg = std::make_shared<UnlabeledValueArg<string>>(mode, "mode is " + mode, true, "", allowedModesConstraint.get());
+    cmdLineParser->add(arg.get());
+    return {arg, allowedModesConstraint};
+}
+
+shared_ptr<Source> ModeCLIParser::processFastqFile(const string &fastqFileVal,
+                                                   const S3ServiceOptions &s3ServiceOptions) {
+    if (fastqFileVal == "-") {      // Streamed mode
```

Comment:
This and the following comment are unnecessary.
```suggestion
    if (fastqFileVal == "-") {
```